File: R/problem_001.R
#!/usr/bin/Rscript

# Original problem
# If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.
#
# Find the sum of all the multiples of 3 or 5 below 1000.
# https://projecteuler.net/problem=1

args <- commandArgs(TRUE)

summation <- function(x) {
	(x*(x+1))/2
}

if(length(args) < 1) {
	print("Please pass an integer from the command line as an argument")
	print("Rscript <<file>> <<arg>>")
	stop()
}

n <- as.numeric(args[1])
if(n %/% 1 != n) {
	print("The value passed to the first argument is not an integer")
	stop()
}

n <- n - 1
div3 <- n %/% 3
div5 <- n %/% 5
div15 <- n %/% 15

result <- summation(div3)*3 + summation(div5)*5 - summation(div15)*15

print(result)

File: java/src/problem_001.java
public class problem_001 {

	public static void main(String[] args) {
		int n = Integer.parseInt(args[0]) - 1;
		System.out.println(summation(n / 3)*3 + summation(n / 5)*5 -
				summation(n / 15)*15);
		
	} //end main
	
	private static int summation(final int n) {
		return (n * (n + 1)) / 2;
	}
} // end problem_001